### PR TITLE
Support inference with RenyiELBO for local latent variable models 

### PR DESCRIPTION
**Pull request by OlaRonning**
Addresses #3104

## TODO
- [x] Add Local latent variable model test case
- [ ] Correct RenyiELBO `logsumexp` reduction dim for LLV model

### CODE REVIEW

**Comment by OlaRonning** [comment]
Should I ignore notebooks when using black? When running `black[jupyter]==22.6.0`, I get 48 files changed.

**Comment by fritzo** [comment]
Yes could you please ignore notebooks?  The easiest way to do so is to run black through `make format`.  Alternatively we could set up a pyproject.toml file with the black exclusion list

**Comment by OlaRonning** [comment]
> Yes could you please ignore notebooks?  

Sure, no problem.

> The easiest way to do so is to run black through `make format`.  Alternatively we could set up a pyproject.toml file with the black exclusion list.

Looking at the format target in the Makefile, it seems it's only black that needs a section in the TOML file. I'll configure a pyproject.toml and update the Makefile.

File: tests/infer/test_inference.py [hunk 1]
```diff
@@ -1005,3 +1005,72 @@ def guide(data, weights):
         loss = svi.step(data, weights)
         if step % 20 == 0:
             logger.info("step {} loss = {:0.4g}".format(step, loss))
+
+
+@pytest.mark.stage("integration", "integration_batch_2")
+class OneWayNormalRandomEffects(TestCase):
```
Comment (martinjankowiak) [inline tests/infer/test_inference.py:1011]: how much time do these tests take?

Comment (OlaRonning) [reply tests/infer/test_inference.py:1011]: > how much time do these tests take?

In CI :
|                                 | Time [sec] |
|---------------------------------|------------|
| `test_renyi_nonreparameterized` | 50.67         |
| `test_renyi_reparameterized`    | 36.59         |
| `test_renyi_vectorized`         | 36.32        |

So all are in the top ten slowest for batch 2.

**Comment by OlaRonning** [comment]
@martinjankowiak, can you suggest a plating structure where we need to compute the dependence (i.e., `all_dims - sum_dims !=  num_particles_vectorized`)? I've been playing around with the current implementation but cannot come up with one.

**Review by martinjankowiak — COMMENTED** [review]
@OlaRonning will this still give results for a local latent variable model like a VAE with a data `plate`? why doesn't this logic sum out the data plate dimension?

**Comment by OlaRonning** [comment]
@martinjankowiak, sorry for the late reply. 
 >  why doesn't this logic sum out the data plate dimension?

From my understanding, this corresponds to including the data plate dimension in [`sum_dims`](https://github.com/pyro-ppl/pyro/blob/dev/pyro/infer/renyi_elbo.py#L107), or have I misunderstood something?
